src/dtls13.c

```diff
@@ -608,28 +600,49 @@ static int Dtls13DetectDisruption(WOLFSSL *ssl, word32 fragOffset)
 
     return 0;
 }
+/* seq is a size 2 array */
+static WC_INLINE void Dtls13MakeRN(
+    word32 epoch, word16 seqHi, word32 seqLo, word32 *seq)
+{
+    seq[0] = (epoch << 16) | (seqHi & 0xFFFF);
+    seq[1] = seqLo;
+}
 
-/* recordNumber is a size 2 array */
-static int Dtls13RtxRecordRecvd(WOLFSSL *ssl, enum HandShakeType hs,
-    Dtls13RtxFSM *fsm, word32 *recordNumber, word32 fragOffset)
+static void Dtls13RtxRemoveCurAck(WOLFSSL *ssl, Dtls13RtxFSM *fsm)
 {
-    int ret;
+    Dtls13RecordNumber *rn, **prev_next;
+    word32 recordNumber[2];
 
-    WOLFSSL_ENTER("Dtls13RtxRecordRecvd");
+    Dtls13MakeRN(ssl->keys.curEpoch, ssl->keys.curSeq_hi, ssl->keys.curSeq_lo,
+        recordNumber);
 
-    ssl->dtls13FastTimeout = 1;
+    prev_next = &fsm->ackRecords;
+    rn = fsm->ackRecords;
 
-    if (!ssl->options.handShakeDone || hs != certificate_request) {
-        /* After handshake we don't save certificate_req in the seen
-           records. This is because we will ack that message with a
-           certificate/cert_verifiy/finished flight and we have not a simple and
-           nice way to remove this from seen records. */
-        ret = Dtls13RtxAddAck(ssl, fsm, recordNumber);
-        if (ret != 0) {
-            WOLFSSL_MSG("can't save ack fragment");
-            return ret;
+    while (rn != NULL) {
+        if (rn->seq[0] == recordNumber[0] && rn->seq[1] == recordNumber[1]) {
+            *prev_next = rn->next;
+            XFREE(rn, ssl->heap, DYNAMIC_TYEP_DTLS_MSG);
+            return;
         }
+
+        prev_next = &rn->next;
+        rn = rn->next;
     }
+}
+
+/* recordNumber is a size 2 array */
+static int Dtls13RtxRecordRecvd(
+    WOLFSSL *ssl, enum HandShakeType hs, Dtls13RtxFSM *fsm, word32 fragOffset)
```

Comment:
Consider renaming this to `Dtls13MsgRecvd` because this doesn't have anything to do with the record layer anymore and it doesn't necessary have to be a rtx msg.

Reply:
So it is now `Dtls13RtxMsgRecvd`. This is definitely pedantic but is the `Rtx` necessary? It is a generic function that doesn't only handle `Rtx`.

Reply:
The `WOLFSSL_ENTER` is now outdated. Please update.